From 0b155ff9a1e19f2b4ed4324e822285d3a667f02a Mon Sep 17 00:00:00 2001
From: Peter Crosthwaite <peter.crosthwaite@xilinx.com>
Date: Mon, 23 Feb 2015 15:04:53 -0800
Subject: [PATCH 13/15] arm: xilinx-zynq-mp-generic: Add external RAM

Zynq MPSoC supports external DDR RAM. Add a RAM at 0 to the model.

Signed-off-by: Peter Crosthwaite <peter.crosthwaite@xilinx.com>
---
 hw/arm/xlnx-zynq-mp-generic.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/hw/arm/xlnx-zynq-mp-generic.c b/hw/arm/xlnx-zynq-mp-generic.c
index ff69b07..7394e82 100644
--- a/hw/arm/xlnx-zynq-mp-generic.c
+++ b/hw/arm/xlnx-zynq-mp-generic.c
@@ -18,9 +18,11 @@
 #include "hw/arm/xlnx-zynq-mp.h"
 #include "hw/boards.h"
 #include "qemu/error-report.h"
+#include "exec/address-spaces.h"
 
 typedef struct XlnxZynqMPGeneric {
     XlnxZynqMPState soc;
+    MemoryRegion ddr_ram;
 } XlnxZynqMPGeneric;
 
 static void xlnx_zynq_mp_generic_init(MachineState *machine)
@@ -36,6 +38,11 @@ static void xlnx_zynq_mp_generic_init(MachineState *machine)
         error_report("%s", error_get_pretty(err));
         exit(1);
     }
+
+    memory_region_init_ram(&s->ddr_ram, NULL, "ddr-ram", machine->ram_size,
+                           &error_abort);
+    vmstate_register_ram_global(&s->ddr_ram);
+    memory_region_add_subregion(get_system_memory(), 0, &s->ddr_ram);
 }
 
 static QEMUMachine xlnx_zynq_mp_generic_machine = {
-- 
2.1.1

